superset/assets/javascripts/explore/AdhocMetric.js

```diff
@@ -23,10 +68,29 @@ export default class AdhocMetric {
 
   equals(adhocMetric) {
     return adhocMetric.label === this.label &&
+      adhocMetric.expressionType === this.expressionType &&
+      adhocMetric.sqlExpression === this.sqlExpression &&
       adhocMetric.aggregate === this.aggregate &&
       (
         (adhocMetric.column && adhocMetric.column.column_name) ===
         (this.column && this.column.column_name)
       );
   }
+
+  isValid() {
+    if (this.expressionType === EXPRESSION_TYPES.SIMPLE) {
+      return !!(this.column && this.aggregate);
```

Comment:
`Boolean()` also works fyi

Reply:
Airbnb js style guide prefer `!!` to `Boolean(...)`- not sure if we have a different style guide for this project, but I like their advice on the whole.

https://github.com/airbnb/javascript#coercion--booleans